pyface/__init__.py

```diff
@@ -26,3 +26,43 @@
     "pyside": ["pyside>=1.2", "pygments"],
     "pyside2": ["pyside2", "shiboken2", "pygments"],
 }
+
+# ============================= Test Loader ==================================
+from os import environ
+from os.path import dirname
+from pyface.util.testing import filter_tests
+from unittest import TestSuite
+
+
+def load_tests(loader, standard_tests, pattern):
+    """ Custom test loading function that enables test filtering using regex
+    exclusion pattern.
+
+    Parameters
+    ----------
+    loader : unittest.TestLoader
+        The instance of test loader
+    standard_tests : unittest.TestSuite
+        Tests that would be loaded by default from this module (no tests)
+    pattern : str
+        An inclusion pattern used to match test files (test*.py default)
+
+    Returns
+    -------
+    filtered_package_tests : unittest.TestSuite
+        TestSuite representing all package tests that did not match specified
+        exclusion pattern.
+    """
+    this_dir = dirname(__file__)
```

Comment:
Perhaps do an `import pyface.toolkit` here to ensure that we have the right toolkit up and running before we try and import anything else.  This should ensure the correct tests actually run.

Reply:
Added the import.

>It might even be possible to use the pyface.toolkit value to select the filter rather than having to rely on an environment variable (but maybe still have the choice overridable by an envirionment variable for safety, if you do this). But isn't needed for this PR.

I tried this out but I think setting the exclusion pattern in `load_tests` makes it very easy to forget that this feature exists. I think it's better if all the configuration for the test run (including exclusion pattern) stays in `etstool.py` where it's easy to find.